### ISSUE TITLE
Support chacha20-poly1305@openssh.com cipher

### ISSUE THREAD

**Issue by tisba** [net-ssh/net-ssh#477]
### Expected behavior

I'm trying to harden SSH on servers I manage. My goal would be able to use

```
KexAlgorithms curve25519-sha256@libssh.org
HostKeyAlgorithms ssh-ed25519
Ciphers chacha20-poly1305@openssh.com
```

Which works very well with `ssh`. For deployments using `net-ssh` via Capistrano, I have to add more (arguably less secure) ciphers.

Cipher `chacha20-poly1305@openssh.com` should be supported.

### Actual behavior

(tested using capistrano)

```
Net::SSH::Exception: could not settle on encryption_client algorithm
```

### System configuration

- net-ssh 4.0
- capistrano 3.7.1
- Ruby 2.3.1/2.3.3/2.4

**Comment by mfazekas** [comment]
Thanks for the report, unfortunately it's not likely to be implemented in the near future unless there is a PR, it sounds it either require some native code or it might be implemented on top of as rbnacl 4.0 seems have it.

See
https://github.com/cryptosphere/rbnacl/blob/master/lib/rbnacl/aead/chacha20poly1305_ietf.rb

See
https://github.com/openssh/openssh-portable/blob/master/PROTOCOL.chacha20poly1305
https://github.com/openssh/openssh-portable/blob/master/chacha.c
https://github.com/openssh/openssh-portable/blob/master/cipher-chachapoly.c

**Comment by fwininger** [comment]
@tisba : I don't have a PR for chacha20-poly1305, but I have one for "KexAlgorithms curve25519-sha256@libssh.org".

**Comment by fwininger** [comment]
relative to #705

**Comment by HLFH** [comment]
Any updates?

```
cap aborted!
SSHKit::Runner::ExecuteError: Exception while executing as deploy@example.com: could not settle on encryption_client algorithm
Server encryption_client preferences: chacha20-poly1305@openssh.com
Client encryption_client preferences: aes256-ctr,aes192-ctr,aes128-ctr


Caused by:
Net::SSH::Exception: could not settle on encryption_client algorithm
Server encryption_client preferences: chacha20-poly1305@openssh.com
Client encryption_client preferences: aes256-ctr,aes192-ctr,aes128-ctr
```

**Comment by fwininger** [comment]
#845

**Comment by mfazekas** [comment]
It's now implemented in 7.2.0.beta1. Please not it's only active if you have RbNaCl gem installed. Pls test and report back.

https://rubygems.org/gems/net-ssh/versions/7.2.0.beta1

**Comment by tisba** [comment]
Looks good in my testing. Not entirely sure what exactly to test, other then I can connect, send commands… it works. Awesome! Thank you! 🙇
